api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -618,4 +620,18 @@
      */
     <T> InterceptionFactory<T> createInterceptionFactory(CreationalContext<T> ctx, Class<T> clazz);
 
+
+    /**
+     *
+     * Obtains an {@link Instance} giving access to beans instances
+     *
+     * Instances of dependent scoped beans obtained with this Instance should be explicitly release with {@link Instance#destroy(Object)}
+     *
+     * @return an {@link Instance} object to request beans instances
+     * @throws IllegalStateException if called during application initialization, before the {@link AfterDeploymentValidation}
+     *         event is fired.
+     * @since 2.0
+     */
+    Instance<Object> getInstance();
```

Comment:
@manovotn I understand your pov, but most of our work in CDI 2.0 has been to add convenient way to do things that were already doable (i.e. configurators api). I think these added value for users are very important because they will bring down the ticket price to write extension and advance CDI code. It will pay in the end.
Regarding the size of BeanManager I agree that it's not the best, but either we have multiple entry point to CDI SPI, either we have something like BM... Remember that JDK8 developers dared to introduce classes like `CompletableFuture` with nearly 80 methods ;)